hoomd/hpmc/ComputeSDF.h

```diff
@@ -364,13 +390,184 @@ template<class Shape> void ComputeSDF<Shape>::countHistogram(uint64_t timestep)
+                                    params[__scalar_as_int(postype_j.w)],
+                                    scale_factor);
+                                if (hard_overlap)
+                                    {
+                                    // add appropriate weight to appropriate bin and exit the
+                                    // loop over bins
+                                    hist_weight_ptl_i = 1.0; // = 1-e^(-\infty)
+                                    min_bin_ptl_i = bin_to_sample;
+                                    break;
+                                    } // end if overlap
+
+                                // if no hard overlap, check for a soft overlap if we have
+                                // patches
+                                if (m_mc->m_patch)
+                                    {
+                                    // compute the energy at this size of the perturbation and
+                                    // compare to the energy in the unperturbed state
+                                    vec3<Scalar> r_ij_scaled = r_ij * (Scalar(1.0) - scale_factor);
+                                    double u_ij_new
+                                        = m_mc->m_patch->energy(r_ij_scaled,
+                                                                typ_i,
+                                                                quat<float>(shape_i.orientation),
+                                                                float(h_diameter.data[i]),
+                                                                float(h_charge.data[i]),
+                                                                typ_j,
+                                                                quat<float>(orientation_j),
+                                                                float(h_diameter.data[j]),
+                                                                float(h_charge.data[j]));
+                                    // if enery has changed, there is a new soft overlap
+                                    // add the appropriate weight to the appropriate bin of the
+                                    // histogram and break out of the loop over bins
+                                    if (u_ij_new != u_ij_0)
+                                        {
+                                        min_bin_ptl_i = bin_to_sample;
+                                        hist_weight_ptl_i = 1.0 - fast::exp(-(u_ij_new - u_ij_0));
+                                        break;
+                                        }
+                                    } // end if (m_mc->m_patch)
+                                }     // end loop over histogram bins
+                            }
+                        }
+                    }
+                else
+                    {
+                    // skip ahead
+                    cur_node_idx += aabb_tree.getNodeSkip(cur_node_idx);
+                    }
```

Comment:
To reduce nesting place this at the top with `aabb_tree.isNodeLeaf(cur_node_idx)`

```python
if (!detail::overlap(aabb_tree.getNodeAABB(cur_node_idx), aabb))
    {
    cur_node_idx += aabb_tree.getNodeSkip(cur_node_idx);
    continue;
    }
```

Reply:
This is just a copy and paste from all the other places that use the AABB tree search in HPMC.